File: BackEnd/XeGo-Microservices/Services/Vehicle/XeGo.Services.Vehicle.API/Entities/Driver.cs
using System.ComponentModel.DataAnnotations;
using XeGo.Shared.Lib.Entities;

namespace XeGo.Services.Vehicle.API.Entities
{
    public class Driver : BaseEntity
    {
        [Key] public string UserId { get; set; } = string.Empty!;
        [Required] public string UserName { get; set; } = string.Empty!;
        [Required] public string FirstName { get; set; } = string.Empty!;
        [Required] public string LastName { get; set; } = string.Empty!;
        [Required] public string PhoneNumber { get; set; } = string.Empty!;
        [Required] public string Email { get; set; } = string.Empty!;
        [Required] public string Address { get; set; } = string.Empty!;
        [Required] public bool IsAssigned { get; set; } = false;
    }
}

File: FrontEnd/xego_rider/lib/models/Entities/driver.dart
class Driver {
  final String userId;
  final String userName;
  final String firstName;
  final String lastName;
  final String phoneNumber;
  final String email;
  final String address;
  final bool isAssigned;
  final String createdBy;
  final String createdDate;
  final String lastModifiedBy;
  final String lastModifiedDate;

  Driver(
      {required this.userId,
      required this.userName,
      required this.firstName,
      required this.lastName,
      required this.phoneNumber,
      required this.email,
      required this.address,
      required this.isAssigned,
      required this.createdBy,
      required this.createdDate,
      required this.lastModifiedBy,
      required this.lastModifiedDate});

  Map<String, dynamic> toJson() => {
        'userId': userId,
        'userName': userName,
        'firstName': firstName,
        'lastName': lastName,
        'phoneNumber': phoneNumber,
        'email': email,
        'address': address,
        'isAssigned': isAssigned,
        'createdBy': createdBy,
        'createdDate': createdDate,
        'lastModifiedBy': lastModifiedBy,
        'lastModifiedDate': lastModifiedDate,
      };

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      userId: json['userId'],
      userName: json['userName'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      phoneNumber: json['phoneNumber'],
      email: json['email'],
      address: json['address'],
      isAssigned: json['isAssigned'],
      createdBy: json['createdBy'],
      createdDate: json['createdDate'],
      lastModifiedBy: json['lastModifiedBy'],
      lastModifiedDate: json['lastModifiedDate'],
    );
  }
}
